plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHttpMetastoreClient.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import io.airlift.testing.Closeables;
+import io.airlift.units.Duration;
+import io.trino.hive.thrift.metastore.Database;
+import io.trino.hive.thrift.metastore.NoSuchObjectException;
+import jakarta.servlet.http.HttpServletRequest;
+import org.apache.http.HttpHeaders;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestThriftHttpMetastoreClient
+{
+    private static final HiveMetastoreAuthentication NO_HIVE_METASTORE_AUTHENTICATION = new NoHiveMetastoreAuthentication();
+    private static final Duration TIMEOUT = new Duration(20, SECONDS);
+    private static InMemoryThriftMetastore delegate;
+    private static TestingThriftHttpMetastoreServer metastoreServer;
+
+    @BeforeClass
+    public static void setup()
+            throws Exception
+    {
+        TestRequestHeaderInterceptor requestHeaderInterceptor = new TestRequestHeaderInterceptor();
```

Comment:
TestRequestHeaderInterceptor is configured in setup and used in testHttpThriftConnection. It makes adding a new test with different headers hard. We could initialize TestingThriftHttpMetastoreServer in the test method.